cdap-security/src/main/java/co/cask/cdap/security/server/ExternalAuthenticationServer.java

```diff
@@ -110,11 +110,21 @@ public ExternalAuthenticationServer(CConfiguration configuration, SConfiguration
    * @return InetSocketAddress of server.
    */
   public InetSocketAddress getSocketAddress() {
-    return new InetSocketAddress(address, port);
+    if (!server.isRunning()) {
```

Comment:
does anybody call this before starting the server?  Would it be better just to add a javadoc that this needs to be called after the server is started, otherwise you can get an invalid address?